### PR TITLE
Query loop format controls: Fix JavaScript error

### LINKED ISSUE #65550: Query Loop block: block error with GB 19.3 RC1 when using block format support
Using GB 19.3RC1 which includes https://github.com/WordPress/gutenberg/pull/64167, I found that after adding support to the TT4 theme, I wasn't able to get the query loop to properly filter with a format option when adding a query loop block to a page or a template. In this case, I had set a few posts to include an Image format in case that matters:

https://github.com/user-attachments/assets/33435441-78aa-4cc5-83fb-91f7fbb1788a

### PR DESCRIPTION
**Pull request by carolinan**
## What?

Fixes https://github.com/WordPress/gutenberg/issues/65550.

This PR fixes a JavaScript error that happens in the `FormatControls` in the query loop block when the
`format` parameter in the `query` block attribute is not an `array`.

This error can occur for example when an **existing** query loop pattern is placed in a post or page, for example
when a block pattern with a query loop is added. Presumably because the 'format' is missing in these existing patterns.

## Why? & How
`FormatControls` already creates a constant called `normalizedFormats` with the purpose of ensuring that `format` is an `array`. - But it was not used in all instances where it was needed:
This PR updates `FormatControls` to use `normalizedFormats` for creating the list of suggested formats in the `FormTokenField`.

## Testing Instructions
First please enable post format support in the active theme, for example in functions.php.
Example:

```
function twentytwentyfour_setup(){
	add_theme_support( 'post-formats', array( 'standard', 'aside', 'gallery', 'audio', 'video', 'link', 'image', 'chat', 'status', 'quote' ) );
}

add_action( 'after_setup_theme', 'twentytwentyfour_setup' );
```

Add some test posts, both with and without post formats assigned.

-**If you are using Twenty Twenty-Four,**  test a page pattern:
Create a new page: You will be prompted with a modal when you create a new page.
In this modal, select the pattern "Portfolio Home with post featured images".

In this portfolio design there are multiple query loop blocks. 
Open the ListView to locate the query loops easier.

Select a query loop and then open the Filters panel in the block settings sidebar.
In the Filters panel, click on "Formats".
Confirm that there are no block validation errors or other issues:
A new input field should show in the Filters panel.
Select the input field, and type or select a post format name.
Confirm that the query loop displays the selected format.

### Test an inserted pattern (any theme)

Create a new post or page.
Open the block inserter in top left corner, and select the Patterns tab.
Select the "Posts" pattern category, and insert any pattern from this list.

Select a query loop and then open the Filters panel in the block settings sidebar.
In the Filters panel, click on "Formats".
Confirm that there are no block validation errors or other issues:
A new input field should show in the Filters panel.
Select the input field, and type or select a post format name.
Confirm that the query loop displays the selected format.

### CODE REVIEW

File: packages/block-library/src/query/edit/inspector-controls/format-controls.js [hunk 1]
```diff
@@ -68,7 +68,7 @@ export default function FormatControls( { onChange, query: { format } } ) {
 		.filter( Boolean );
 
 	const suggestions = formats
-		.filter( ( item ) => ! format.includes( item.value ) )
+		.filter( ( item ) => ! normalizedFormats.includes( item.value ) )
```
Comment (Mamaduka) [inline packages/block-library/src/query/edit/inspector-controls/format-controls.js:71]: Just to refresh my memory, this filter removes values from the suggestions already assigned to the Query, correct?

**Review by Mamaduka — APPROVED** [review]
Thank you, @carolinan!

The fix tests well for me ✅

File: packages/block-library/src/query/edit/inspector-controls/format-controls.js (hunk 1 above)
Comment (carolinan) [reply packages/block-library/src/query/edit/inspector-controls/format-controls.js:71]: Yes.
Shall I add a comment about it? I still need to create a new PR to apply the changes requested in the review of the sync with core, I could add it then.

**Comment by kevin940726** [comment]
I just cherry-picked this PR to the release/19.3 branch to get it included in the next release: 35f02012107

File: packages/block-library/src/query/edit/inspector-controls/format-controls.js (hunk 1 above)
Comment (Mamaduka) [reply packages/block-library/src/query/edit/inspector-controls/format-controls.js:71]: There's no need for a comment. I was just double checking for myself 😄

**Pull request merged**